packages/cdktf-cli/lib/server/server.ts

```diff
@@ -0,0 +1,136 @@
+import express from "express";
+import { createServer } from "http";
+import { ApolloServer } from "apollo-server-express";
+import { execute, subscribe } from "graphql";
+import { ApolloServerPluginLandingPageGraphQLPlayground } from "apollo-server-core";
+import { SubscriptionServer } from "subscriptions-transport-ws";
+import { loadSchema } from "@graphql-tools/load";
+import { GraphQLFileLoader } from "@graphql-tools/graphql-file-loader";
+import { PubSub } from "graphql-subscriptions";
+import path from "path";
+import { WatchClient, WatchState } from "./WatchClient";
+import { GraphQLWatchState, mapWatchState } from "./util";
+
+type WatchInputs = {
+  targetDir: string;
+  synthCommand: string;
+  autoApprove: boolean;
+  targetStack?: string;
+};
+
+const pubsub = new PubSub();
+const WATCH_EVENT = "WATCH";
+
+const PORT = process.argv[2];
+
+console.log("server started with cwd:", process.cwd());
+
+let watchClient: WatchClient | undefined;
+
+async function startApolloServer() {
+  // Provide resolver functions for your schema fields
+  const resolvers = {
+    Query: {
+      hello: () => "Hello world!!!",
+    },
+    Subscription: {
+      watch: {
+        subscribe: async (
+          _parent: undefined,
+          args: { inputs: WatchInputs }
+        ) => {
+          // create this here so that events with errors published
+          // while startup reach the client
+          const iterator = pubsub.asyncIterator([WATCH_EVENT]);
+
+          if (watchClient && (await watchClient.isRunning())) {
+            console.warn(
+              "There is already a watch running, you will receive the updates from that run but no new one is started."
+            );
+          } else {
+            try {
+              watchClient = new WatchClient(args.inputs);
+
+              // TODO: subscribe returns an unsubscribe function but that connect reliably
```

Comment:
I show how to use RxJS instead in this video, but that might be overkill: https://www.youtube.com/watch?v=7SOwFE9lU-g

Reply:
Hehe, neat idea – but working with Observables would be another concept that we'd introduce in our codebase then 😄 But maybe that's a neat way of doing that at some point – who know 😁

Reply:
:D yeah not really worth it here. But it also explains how to implement your own async iterable which could be useful here 🤔

Reply:
Ah, nice! Gonna resolve this for now anyway. bye 👋😄

Reply:
Oh, it was resolved already